README.md

```diff
@@ -99,30 +99,35 @@ Contact: firstname.lastname@hpi.de
 
```

Comment:
(Random line) @mrks @Felerius 
> @mrzzzrm Going back to what we discussed here: I think I found a case where we need to crawl subquery LQPs of predicate nodes we check:
`SELECT * FROM a WHERE a.a IN (SELECT b.b FROM b WHERE EXISTS (SELECT c.c IN C WHERE c.c = a.a))`

I don't think you should need to look at `SELECT c.c IN C WHERE c.c = a.a`. The PredicateNode processing `EXISTS (SELECT c.c IN C WHERE c.c = a.a))` will have a parameter `a.a` in its `predicate`. There should never be a need for you to go into `SELECT c.c IN C WHERE c.c = a.a`, however deeply nested of a subquery it might be.

> For some reason, there are cross joins emitted here.
@mrzzzrm - any idea why this passed all tests?

Because... there are just no tests for the JoinOrderingRule pulling predicates past Semi/Anti/Outer joins.

> Depends on how hard that is to check. For inner joins it should be relatively easy (I think it might even be always possible for the kind of predicates that we pull up, but I'm not quite sure yet), however for outer joins I'm unsure about how nullability needs to be handled.

You can always pull up predicates past Inner Joins, from the outer side of a left/right outer join and from the left side of a semi/anti join.
Both rules could be blamed here: The JoinOrderingRule (more specifically: the JoinGraph(Builder)) should be able to "look" past Semi/Anti joins, the SubqueryToJoinRule should be able to look past inner joins. For the moment, I think the SubqueryToJoinRule should be run before the JoinOrderingRule and should be fixed to be able to pull up predicates past inner joins, which is really easy (at least easier than what we'd need to do to JoinGraph and its builder in order to make it work with non-inner joins)